### PR TITLE
NV-2136 - 🐛 Bug Report: Novu Provider is not working properly.

### PR DESCRIPTION
**Pull request by gitstart**
### What change does this PR introduce?
 Refetches organizations info, unseen count and notifications feeds when `subscriberId` or `applicationIdentifier` changes.

 ### Why was this change needed?
Closes: https://github.com/novuhq/novu/issues/3234

### Other information

---
This code was written and reviewed by GitStart Community. Growing great engineers, one PR at a time.

### CODE REVIEW

**Comment by gitstart** [comment]
@scopsy this PR is ready for review

File: packages/notification-center/src/hooks/useFetchOrganization.ts [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
+  const [intialize, setInitialize] = useState(false);
+  const queryClient = useQueryClient();
+
+  useEffect(() => {
+    if (intialize) {
+      queryClient.refetchQueries(UNSEEN_COUNT_QUERY_KEY, { exact: false });
+      queryClient.refetchQueries(FEED_UNSEEN_COUNT_QUERY_KEY, { exact: false });
+      queryClient.refetchQueries(ORGANIZATION_QUERY_KEY, { exact: false });
+    }
+  }, [subscriberId, applicationIdentifier]);
```
Comment (LetItRock) [inline packages/notification-center/src/hooks/useFetchOrganization.ts:21]: This hook is supposed to be only responsible for fetching the organization. Now it has additional logic that refetches other queries after we fetched org. I think this is not a good place for this logic to be and I wouldn't expect that if I will debug it. And why do we only refetch these and not for ex. the user preferences?

---

I haven't spend much time digging into the code, but I feel like we should do this in another way. 
In simple words, we just need to update the "query keys" with the `subscriberId, subscriberHash, applicationIdentifier` like the `useSession` does, so any changes to these fields will create the new "query keys" resulting in new fetches for all queries (if the cache is stale).

Please ensure that all the query hooks have these fields in the keys and that the WS session is also re-initialized.

**Comment by gitstart** [comment]
@LetItRock thanks for the feedback, your comment has been implemented.

**Review by LetItRock — APPROVED** [review]
amazing work! 🥇

**Comment by jainpawan21** [comment]
@gitstart few tests are failing.
Can you please look into these failing tests?

File: packages/notification-center/src/hooks/useInitializeSocket.ts [hunk 2]
```diff
@@ -11,6 +12,7 @@ type IUseInitializeSocket = (args: { socketUrl: string }) => {
 
 export const useInitializeSocket: IUseInitializeSocket = ({ socketUrl }) => {
   const socketRef = useRef<Socket | null>(null);
+  const setQueryKey = useSetQueryKey();
```
Comment (ainouzgali) [inline packages/notification-center/src/hooks/useInitializeSocket.ts:15]: Hi @gitstart I believe your problem is coming from here. `useInitializeSocket` is used in NovuProvider.ts but `useSetQueryKey` uses the context of that provider.

Comment (gitstart) [reply packages/notification-center/src/hooks/useInitializeSocket.ts]: I have effected the changes @ainouzgali

**Review by ainouzgali — APPROVED** [review]
Thank you @gitstart ! This looks great!

**Comment by donnguyen** [comment]
Hi guys, just wonder when will this fix be released?

Thanks again for the great work!

**Comment by ainouzgali** [comment]
Hi @donnguyen, this was included in the release of v0.16.0 which was deployed 2 days ago :) 
Could you please try it out and let us know if it works out for you?

**Comment by donnguyen** [comment]
Hi @ainouzgali, thanks for getting back to me. I have just tried the new version, but seems it hasn't solved the whole issues yet.

The issue now I can find is that the notification/organization/unseen calls might happen before the initialise (where we get and set token), and if the case happens -> these API calls still get data of the old/previous subscriber/application.

**Pull request merged**